src/Compilers/Core/AnalyzerDriver/AnalyzerExecutor.cs

```diff
@@ -79,15 +71,13 @@ internal class AnalyzerExecutor
             Compilation compilation,
             AnalyzerOptions analyzerOptions,
             Action<Diagnostic> addDiagnostic,
```

Comment:
addDiagnostic is for registering a diagnostic reported by analyzer.
onAnalyzerException is for handling analyzer exception and corresponding diagnostic generated by the driver.